doc/development/plugins.rst

```diff
@@ -232,6 +232,68 @@ or can include in-built transforms such as:
 * :func:`pennylane.devices.preprocess.validate_multiprocessing_workers`
 * :func:`pennylane.devices.preprocess.validate_adjoint_trainable_params`
 * :func:`pennylane.devices.preprocess.no_sampling`
+  
+**Custom decompositions:**
+
+When adding a new pluging, one may need to define how unsupported operations decompose to the
+device's native gates. The transform can be specified via the decomposer keyword argument of
+:func:`pennylane.devices.preprocess.decompose`.
+
+For example, suppose we have a device with RX, RY and IsingXX as native gates but we want to execute a circuit
+written in terms of CNOTs. First, we need to define a decomposition function (e.g., `decompose_cnot`) that takes
+the CNOT operator as argument:
+
+.. code-block:: python
+    
+    def decompose_cnot(op: qml.CNOT):
+        wires = op.wires
+        return [
+            qml.RY(np.pi/2, wires=wires[0]),
+            qml.IsingXX(np.pi/2, wires=wires),
+            qml.RX(-np.pi/2, wires=wires[0]),
+            qml.RY(-np.pi/2, wires=wires[0]),
+            qml.RY(-np.pi/2, wires=wires[1])
+        ]
+
+Then, we can add the transform with:
+
+.. code-block:: python
+
+    from pennylane.devices.preprocess import decompose
+
+    def supports_operation(op): 
+        return getattr(op, "name", None) in operation_names
+        
+    program.add_transform(decompose, stopping_condition=supports_operation, decomposer=decompose_cnot, name="my_device")
```

Comment:
Actually, the decomposer overrides `op.decomposition()` for all ops, so it'll need to be something like
```python
def custom_decomposer(op):
    if isinstance(op, qml.CNOT):
        return decompose_cnot(op)
    return op.decomposition()

program.add_transform(
    decompose,
    stopping_condition=supports_operation,
    decomposer=custom_decomposer,
    name="my_device"
)
```